elf/elf.go

```diff
@@ -293,31 +286,69 @@ func elfReadVersion(file *elf.File) (uint32, error) {
 	return 0, nil
 }
 
-func prepareBPFFS(namespace, name string) (string, error) {
+func createPinPath(path string) (string, error) {
 	err := bpffs.Mount()
 	if err != nil {
 		return "", err
 	}
-	mapPath := filepath.Join(BPFFSPath, namespace, BPFDirGlobals, name)
-	err = os.MkdirAll(filepath.Dir(mapPath), syscall.S_IRWXU)
+	err = os.MkdirAll(filepath.Dir(path), syscall.S_IRWXU)
 	if err != nil {
-		return "", fmt.Errorf("error creating map directory %q: %v", filepath.Dir(mapPath), err)
+		return "", fmt.Errorf("error creating map directory %q: %v", filepath.Dir(path), err)
+	}
+	return path, nil
+}
+
+func validateMapPath(path string) bool {
+	relPath := strings.TrimPrefix(path, BPFFSPath)
```

Comment:
Can you also check if the path actually starts with `BPFFSPath` and add a test for that?